bin/setup.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/sh
+
+# Setup script for Artsy's use.
+
+# It assumes that the general Artsy pre-requisites are already installed.
+# Please see: https://github.com/artsy/potential/blob/master/scripts/setup
+
+trap 'ret=$?; test $ret -ne 0 && printf "Setup script failed.\n\n" >&2; exit $ret' EXIT
+
+set -e
+
+python_version=$(cat .python-version)
+if (command -v pyenv >/dev/null)
+then
+  echo "Found Pyenv."
+  if (pyenv versions | grep -Fq "$python_version")
+  then
+    echo "Found Python $python_version."
+  else
+    echo "Using Pyenv to install Python $python_version..."
+    pyenv install -s "$python_version"
+  fi
+  pyenv rehash
+  echo "Installing Poetry..."
+  pip install poetry
+  echo "Running poetry install..."
+  poetry install
```

Comment:
I think you probably need to activate a virtualenv here, otherwise you're installing these packages globally _for the given version of python_ you're using via pyenv. I don't tend to use `rehash` but I don't think that's going to isolate packages to this project... we generally want project specific environments. You can see how we do that in other projects [here](https://github.com/artsy/degas/blob/main/scripts/setup.sh#L53-L67).

Reply:
yeah, does look weird not having `pyenv activate ...`. well, now i have copied degas setup script, with some edits. the rehash is retained, as without it, it cannot find `pre-commit` command.